parsers/ESIOS.py

```
@@ -0,0 +1,62 @@
+# The arrow library is used to handle datetimes
+import arrow
+import urllib
+# The request library is used to fetch content through HTTP
+import requests
+from os import environ
+
+
+def fetch_exchange(country_code1='ES', country_code2='MA', session=None):
+
+    ## Get ESIOS token
+    token = environ.get('ESIOS_TOKEN', "")
+    if not token:
+        print '[ERROR]: ESIOS Parser require access token'
+        return None
+
+    ses = session or requests.Session()
+
+    ## Request headers
+    headers = {'Content-Type': 'application/json',
+               'Accept': 'application/json; application/vnd.esios-api-v2+json',
+               'Authorization': 'Token token="{0}"'.format(token)}
+
+    ## Request query url
+    utc = arrow.utcnow()
+    start_date = utc.shift(hours=-1).floor('hour').isoformat()
+    end_date = utc.ceil('hour').isoformat()
+    dates = {'start_date': start_date, 'end_date': end_date}
+    query = urllib.urlencode(dates)
+    url = 'https://api.esios.ree.es/indicators/10209?{0}'.format(query)
+
+    response = ses.get(url, headers=headers)
+    if response.status_code != 200 or not response.text:
+        print '[WARNING]: ESIOS Parser Response code:', response.status_code
```

can you throw an exception instead? Same for the other cases.